src/tools/tidy/src/rustdoc_html_ids.rs

```diff
@@ -0,0 +1,164 @@
+//! Checks that the rustdoc ID map is up-to-date. The goal here is to check a few things:
+//!
+//! * All IDs created by rustdoc (through JS or files generation) are declared in the ID map.
+//! * There are no unused IDs.
+
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+
+use regex::Regex;
+
+const ID_MAP_PATH: &str = "librustdoc/html/markdown.rs";
+const IDS_USED_IN_JS: &[&str] = &[
+    // This one is created in the JS and therefore cannot be found in rust files.
+    "help",
+];
+
+fn extract_ids(path: &Path, bad: &mut bool) -> HashMap<String, usize> {
+    let file = File::open(path).expect("failed to open file to extract rustdoc IDs");
+    let buf_reader = BufReader::new(file);
+    let mut iter = buf_reader.lines();
+    let mut ids = HashMap::new();
+
+    while let Some(Ok(line)) = iter.next() {
+        if line.trim_start().starts_with("ID_MAP!(") {
```

Comment:
This would need to be updated as well:

```suggestion
        if line.trim_start().starts_with("id_map!(") {
```